crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -437,11 +480,13 @@ y = 2
 ",
         );
         let index = semantic_index(&db, file);
-        let root_table = index.symbol_table(FileScopeId::root());
+        let module_global_table = index.symbol_table(FileScopeId::module_global());
 
-        assert_eq!(names(&root_table), vec!["func", "y"]);
+        assert_eq!(names(&module_global_table), vec!["func", "y"]);
 
-        let scopes = index.child_scopes(FileScopeId::root()).collect::<Vec<_>>();
+        let scopes = index
+            .child_scopes(FileScopeId::module_global())
```

Comment:
I somewhat strongly prefer module-global. The topmost scope is always called the "module" scope or the "global" scope in a Python context; I'm not sure I've ever heard it referred to as the "root" scope.

Reply:
I think I then prefer `module` or `global`. Using both seems unnecessary complex.

Reply:
I think this discussion clarifies why "root scope" is a bad term for module-global scope, because it is not actually the root scope in terms of lookup order; builtins scope is the root scope in that sense (the last fallback for name lookups.) All scopes are "children" of the builtins scope in that sense.

I'm OK with just using "module scope" for this, I think that's clear enough.

Reply:
Hmm, now I remember why I went with `module_global` and not just `module` -- because we have cases like `module_global_symbol_ty`, and changing that to `module_symbol_ty` is confusing, because we have some symbols that represent an entire module (like `foo` in `import foo`), and "module symbol" sounds like it could be that, rather than "a module-scope symbol".

I think just "global" could be OK too, as long as we understand that in Python, "global" means "module global" (but in Python that is what it generally means.)

I might leave this change for a follow-up PR, because it's not feeling clear to me.

Reply:
https://github.com/astral-sh/ruff/pull/12374